tests/edu/oregonstate/cs361/battleship/MainTest.java

```diff
@@ -61,9 +61,30 @@ public void testScan() {
+        assertEquals("{\"clipper\":{\"name\":\"Clipper\",\"length\":3,\"start\":{\"Across\":0,\"Down\":0},\"end\":{\"Across\":0,\"Down\":0}},\"dinghy\":{\"name\":\"Dinghy\",\"length\":1,\"start\":{\"Across\":0,\"Down\":0},\"end\":{\"Across\":0,\"Down\":0}},\"aircraftCarrier\":{\"stealth\":false,\"name\":\"AircraftCarrier\",\"length\":5,\"start\":{\"Across\":0,\"Down\":0},\"end\":{\"Across\":0,\"Down\":0}},\"battleship\":{\"stealth\":true,\"name\":\"Battleship\",\"length\":4,\"start\":{\"Across\":0,\"Down\":0},\"end\":{\"Across\":0,\"Down\":0}},\"submarine\":{\"stealth\":true,\"name\":\"Submarine\",\"length\":2,\"start\":{\"Across\":0,\"Down\":0},\"end\":{\"Across\":0,\"Down\":0}},\"computer_clipper\":{\"name\":\"Computer_Clipper\",\"length\":3,\"start\":{\"Across\":5,\"Down\":1},\"end\":{\"Across\":5,\"Down\":3}},\"computer_dinghy\":{\"name\":\"Computer_Dinghy\",\"length\":1,\"start\":{\"Across\":10,\"Down\":10},\"end\":{\"Across\":10,\"Down\":10}},\"computer_aircraftCarrier\":{\"stealth\":false,\"name\":\"Computer_AircraftCarrier\",\"length\":5,\"start\":{\"Across\":2,\"Down\":2},\"end\":{\"Across\":2,\"Down\":6}},\"computer_battleship\":{\"stealth\":true,\"name\":\"Computer_Battleship\",\"length\":4,\"start\":{\"Across\":2,\"Down\":8},\"end\":{\"Across\":5,\"Down\":8}},\"computer_submarine\":{\"stealth\":true,\"name\":\"Computer_Submarine\",\"length\":2,\"start\":{\"Across\":9,\"Down\":6},\"end\":{\"Across\":9,\"Down\":7}},\"playerHits\":[],\"playerMisses\":[],\"computerHits\":[],\"computerMisses\":[],\"shipSquares\":[],\"scanResult\":false,\"overlapResult\":false,\"offBoard\":false,\"snakeRow\":1,\"snakeCol\":1}",res.body);
     }
 
+    @Test
```

Comment:
Yay, verified that we're hanging on to 90% line coverage! Concerning these two tests that test mode, they probably contribute positively to the coverage as well but functionally they are only asserting that 1) the post request made it through and 2) the JSON-ified model made it through. But since the easy and hard mode models are identical JSON-structure wise, it cannot be concretely verified that we received the specific type of game instance that we were expecting. To go one step further... I guess you would grab the actual Java object?? and make sure that it is either ``BattleshipModel`` or ``HardBattleshipModel`` with something like ``.getClass()``?? https://docs.oracle.com/javase/tutorial/reflect/class/classNew.html